src/CMakeLists.txt

```diff
@@ -21,12 +21,14 @@ set(CMAKE_ASM_COMPILER  ${PROJECT_SOURCE_DIR}/src/yasm-wrapper)
 set(CMAKE_ASM_FLAGS "-f elf64")
 set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -rdynamic -Wall -Wtype-limits -Wignored-qualifiers -Winit-self -Wpointer-arith -Werror=format-security -fno-strict-aliasing -fsigned-char -fPIC")
 
+if(NOT CMAKE_BUILD_TYPE) 
+  set(CMAKE_BUILD_TYPE "RelWithDebInfo" CACHE STRING "Default BUILD_TYPE is RelWithDebInfo, other options are: Debug Release MinSizeRel.") 
+endif(NOT CMAKE_BUILD_TYPE)
+
 include(CheckCCompilerFlag)
-CHECK_C_COMPILER_FLAG("-Wp,-D_FORTIFY_SOURCE=2" HAS_FORTIFY_SOURCE)
-if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
+CHECK_C_COMPILER_FLAG("-Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=2 -Werr" HAS_FORTIFY_SOURCE)
```

Comment:
@alimaredia The compiler check fails:

/usr/bin/cc    -DHAS_FORTIFY_SOURCE   -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=2 -Werr -o CMakeFiles/cmTC_b97c1.dir/src.c.o   -c /home/build/ceph/test-cmake/build/CMakeFiles/CMakeTmp/src.c
cc: error: unrecognized command line option '-Werr'
make[1]: *** [CMakeFiles/cmTC_b97c1.dir/build.make:66: CMakeFiles/cmTC_b97c1.dir/src.c.o] Error 1
make[1]: Leaving directory '/home/build/ceph/test-cmake/build/CMakeFiles/CMakeTmp'
make: *** [Makefile:126: cmTC_b97c1/fast] Error 2

Changing -Werr to -Werror fixes it. 

Thanks!